proto-grpc.md

```diff
@@ -0,0 +1,277 @@
+  comfortable overriding dependencies. This also makes maintenance of these rulesets more
+  challenging, since rules authors must keep on top of updates to their upstream dependencies.
+
+  The introduction of [bzlmod](https://docs.bazel.build/versions/main/bzlmod.html) to Bazel 
+  (expected in version 6, but testable with an experimental flag in version 5) will hopefully
+  resolve many of the dependency issues listed above and in doing so remove workload from rules
+  authors and allow rules users more freedom to choose versions. Under bzlmod, the dependency tree
+  is explicitly managed by the end user, with updates to transitive dependencies handled
+  transparently without updates needed to intermidiary rulesets.
+
+The above four points dictate much of fragmentation seen between rulesets, since there are numerous
+solutions to these problems, each with their own tradeoffs. The solution that works well for one
+language may be less suitable for another, leading to naturally diverging behaviour. As such,
+hopefully it is now slightly cleared why there is such a diversity of Protobuf and gRPC rulesets,
+with no single 'right answer' that solves everyones' problems.
+
+
+## Custom Plugins
+
+Once you've got Protobuf and gRPC fully integrated with your project and workspace, you may then
+find yourself wanting to use a custom protoc plugin that is either closed source or is otherwise
+unsupported by any of the available rulesets. In these cases, rather than completely reinvent the
+protoc calling conventions in your own rules, both of the cross-language rulesets linked above
+provide mechanisms for integrating your plugin with their standard calling convention.
+
+With stackb/rules_proto, you can
+[extend the Gazelle plugin](https://github.com/stackb/rules_proto/tree/master/pkg/plugin) to support
+your custom plugin's, by capturing its full behaviour in a few Go functions. With rules_proto_grpc,
+you can
+[define your own](https://rules-proto-grpc.com/en/latest/custom_plugins.html) `proto_plugin` and the
+supporting compilation rules.
+
+There are situations in which these tools are insufficient for a specific plugin, but many of the
+more standard use-cases will be supported here.
+
+
+## Common Issues
+
+When working with Protobuf and gRPC under Bazel, there are some issues that are widely seen by many
+users:
+
+- **Caching**: The way Protobuf is built with Bazel is incredibly fragile to cache misses, leading
+  to frequent rebuilds of protoc for no clear reason. This issue is particularly prevalent on CI
+  systems, where getting protoc to cache is particularly challenging. There is no clear single
+  cause for this behaviour, since apparently identical environments may still miss the cache.
+
+  This has led to protoc being downloaded as a prebuilt binary in bazelbuild/rules_proto, but this
+  mechanism breaks when a user wants to use a different protobuf version.
```

Comment:
We should mention that current `rules_proto` gives you a pre-built `protoc`.

Reply:
stackb/rules_proto or bazelbuild/rules_proto?

Reply:
bazelbuild

Reply:
This is mentioned on line 254? Or are you thinking something more specific?